examples-zio2/src/main/scala/tofu/example/logging/impl/SimpleContext.scala

```diff
@@ -0,0 +1,23 @@
+package tofu.example.logging.impl
+
+import tofu.logging.{Loggable, LoggedValue}
+import tofu.logging.zlogs.ContextProvider
+import zio.{FiberRef, UIO, ULayer, ZLayer}
+
+class SimpleContext(ref: FiberRef[String]) extends ContextProvider {
+  private implicit val requestIdLoggable: Loggable[String] =
+    Loggable.stringValue.named("requestId")
+
+  override def getCtx: UIO[LoggedValue] = ref.get.map(x => x)
```

Comment:
Is there any other way to get implement this method other than that? It feels like any implementation of `ContextProvider` will be doing `something.something.map(x => /*implicit conversion*/ x)`. Perhaps it's better to make it `ContextProvider[A: Loggable]`?

Reply:
`ZLogs.layerPlainWithContext` is pretty now: we can provide them any implementation of `ContextProvider`.
If we make this trait as `ContextProvider[A: Loggable]`, the layer will also require  type-parameter which can be huge enough. For example, next PR will bring `ContextProvider` with data `A = Map[ZLogAnnotation[_], Any]`. 

As a user, I don't want write THIS type `A`, ideally I should not know anything about it. I just write two things
```
zioProgram.provide(
  ZLogs.layerPlainWithContext,  // dear tofu, give me a logger factory using ContextProvider I point next to
  TofuDefaultContext.layer         // I'd like to log this context, don't carry how this collected
)
```

Reply:
> will be doing something.something.map(x => /*implicit conversion*/ x)

suggest `abstract class ValueContextProvider` which do this itself

Reply:
> suggest abstract class ValueContextProvider which do this itself

yep, that'll work